blockly/blocks/component_variable.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2012 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Variable input field of a specific component type.
+ * @author fraser@google.com (Neil Fraser)
+ */
+'use strict';
+
+goog.provide('Blockly.Blocks.ComponentFieldVariable');
+
+goog.require('Blockly.Blocks');
+//goog.require('Blockly.FieldVariable');
```

Comment:
I saw other block files which depend on not mentioned requires which are present due to closure parsing, so commented it out. Originally I tried to have this present in blocks/arduino/ which is not possible due to dependencies, then this require is needed